src/targets/gpu/hip_gemm_impl.cpp

```diff
@@ -472,6 +473,53 @@ struct hip_gemm_impl
         return solution_idx;
     }
 
+    /**
+     * Get workspace size for the solution index:  Gets algo from the solution index,
+     * and calls matmulIsAlgoSupported() to get the workspace size.
+     */
+
+    size_t get_workspace_size(context& ctx,
+                              const std::vector<shape>& input_shapes,
+                              int32_t solution_idx) const
+    {
+        size_t workspace_size = hipblaslt_workspace_size;
+        std::vector<argument> input_args;
+        std::transform(input_shapes.begin(),
+                       input_shapes.end(),
+                       std::back_inserter(input_args),
+                       [](const shape& x) { return to_gpu(generate_argument(x)); });
+
+        std::vector<int32_t> algo_index = {solution_idx};
+        std::vector<hipblasLtMatmulHeuristicResult_t> heuristic_result;
+
+        hipblaslt_invoke([&]() {
+            return hipblaslt_ext::getAlgosFromIndex(
+                ctx.get_stream().get_hipblaslt(), algo_index, heuristic_result);
+        });
+        assert(heuristic_result.size() == 1);
```

Comment:
Why are we concerned if we get back more than one result? Is there some metric to pick between the two?

Reply:
Just an extra check to make sure we get one algo from the index that we have provided. This would also trigger if we do not get any algo, and can be helpful in debugging, if it ever happens.

Reply:
All good